core/interfaces/i_toolbox.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a toolbox.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IToolbox');
+
+goog.require('Blockly.IDisposable');
+goog.require('Blockly.IRegistrable');
+
+
+/**
+ * @extends {Blockly.IDisposable}
+ * @extends {Blockly.IRegistrable}
+ * @interface
+ */
+Blockly.IToolbox = function() {};
+
+/**
+ * Initializes the toolbox.
+ */
+Blockly.IToolbox.prototype.init;
+
+/**
+ * Fill the toolbox with categories and blocks.
+ * @param {Array.<Blockly.utils.toolbox.Toolbox>} toolboxDef Array holding objects
+ *    containing information on the contents of the toolbox.
+ */
+Blockly.IToolbox.prototype.render;
+
+/**
+ * Dispose of this toolbox.
+ */
+Blockly.IToolbox.prototype.dispose;
```

Comment:
I would imagine this would already be in `Blockly.IDisposable`, so there's no need to repeat it here.